src/app/index.js

```diff
@@ -109,19 +114,71 @@ export default class extends Generator {
 
     this.registerTransformStream(
       rename(path => {
+        const extension = this.props.useTypeScript ? '.ts' : '.js'
         if (path.basename === 'index' && !path.extname) {
-          path.extname = '.js'
+          path.extname = extension
         }
 
         if (path.extname === '.test') {
           path.basename += '.test'
-          path.extname = '.js'
+          path.extname = extension
         }
 
         return path
       })
     )
 
+    const platformProperties = {}
+    if (this.props.useTypeScript) {
+      platformProperties[
+        'scripts'
+      ] = Object.assign(defaultPackageJson.scripts, {
+        build: 'tsc',
```

Comment:
A new TypeScript danger plugin has a `tslint.json`, but there is no "lint" script. Thoughts on adding this npm script:

```
"lint": "tslint 'src/**/*.ts'
```